packages/embed/src/js/metadata/index.js

```diff
@@ -0,0 +1,235 @@
+ * event to mark the end of the process, calculates the duration of the
+ * process, and stores the duration in the metadata object.
+ */
+export const endTrackDocumentFrontCapture = () => {
+  if (!captureStartTimestamp) {
+    return;
+  }
+  metadata.document_front_capture_duration_ms =
+    Date.now() - captureStartTimestamp;
+  captureStartTimestamp = null;
+  capturing = null;
+
+  console.log(metadata);
+};
+
+/**
+ * Start tracking the document back capture process. This function should be
+ * called once the user starts the document back capture process. It sets the
+ * document_back_capture_retries to 0 and document_back_image_origin to
+ * 'camera'.
+ */
+export const beginTrackDocumentBackCapture = () => {
+  if (metadata.document_back_capture_retries === null) {
+    metadata.document_back_capture_retries = 0;
+  }
+  metadata.document_back_image_origin = 'camera';
+  capturing = 'document_back';
+  if (activeCameraName) {
+    metadata.document_back_capture_camera_name = activeCameraName;
+  }
+  captureStartTimestamp = Date.now();
+};
+
+const retryDocumentBackCapture = () => {
+  metadata.document_back_capture_retries += 1;
+};
+
+/**
+ * End tracking the document back capture process. This function should be
+ * called once the document back capture process is finished. It logs an
+ * event to mark the end of the process, calculates the duration of the
+ * process, and stores the duration in the metadata object.
+ */
+export const endTrackDocumentBackCapture = () => {
+  if (!captureStartTimestamp) {
+    return;
+  }
+  metadata.document_back_capture_duration_ms =
+    Date.now() - captureStartTimestamp;
+  captureStartTimestamp = null;
+  capturing = null;
+  console.log(metadata);
+};
+
+/**
+ * Start tracking the selfie capture process. This function should be called
+ * once the user starts the selfie capture process. It sets the
+ * selfie_image_origin to 'camera'.
+ */
+export const beginTrackSelfieCapture = () => {
+  metadata.selfie_image_origin = 'camera';
+  capturing = 'selfie';
+  if (activeCameraName) {
+    metadata.camera_name = activeCameraName;
+  }
+  captureStartTimestamp = Date.now();
+};
+
+/**
+ * End tracking the selfie capture process. This function should be called
+ * once the selfie capture process is finished. It logs an event to mark
+ * the end of the process, calculates the duration of the process, and
+ * stores the duration in the metadata object.
+ */
+export const endTrackSelfieCapture = () => {
+  if (!captureStartTimestamp) {
+    return;
+  }
+  metadata.selfie_capture_duration_ms = Date.now() - captureStartTimestamp;
+  captureStartTimestamp = null;
+  capturing = null;
+  console.log(metadata);
+};
+
+window.addEventListener('metadata:initialize', initializeMetadata);
+// Document Front
+window.addEventListener(
+  'metadata:document-front-capture-start',
+  beginTrackDocumentFrontCapture,
+);
+window.addEventListener(
```

Comment:
It'll be good to scope the events to our internal elements to reduce external influence.

eg. instead of `window.addEvent('metadata:document-front-origin')` we have  `smartCameraWeb.addEvent('metadata:document-front-origin')`